### DISCUSSION TITLE
JS/Typescript Backend for Tauri

### DISCUSSION THREAD

**Discussion by FleetAdmiralJakob** [tauri-apps/tauri#6891] [Q&A]
I don't know Rust, but I love Typescript, explicitly with Next.js. Is there a way to run my Tauri Backend with Typescript or the Next.js backend features?

**Comment by simonhyll** [comment]
Nope. You can sidecar node.js into your application and run code there, but Tauri is a Rust project, if you want to develop backend functionality you'll have to use Rust. You can get pretty far with just using the Tauri JS API though and plugins. But Tauri doesn't provide any form of javascript runtime that you can tap into.

**Reply by badalya1** [reply]
Is it viable to run a node process and proxy everything through tauri?

**Reply by FleetAdmiralJakob** [reply]
Can't I call my next.js apis from Tauri and only use JS/TS?
